src/plugins/data/common/search/search_source/types.ts

```diff
@@ -29,6 +29,7 @@
  */
 
 import { NameList } from 'elasticsearch';
+import { DataSource } from 'src/plugins/data/public';
```

Comment:
does OSD complain about this? common folder should prohibit stuff from public and server folders.

however, I think this raises a good point that this data source should be within the common folder. if it is in conflicts with the other data_sources perhaps a different name for it temporarily and aliasing it so we don't have to change in many locations.

Reply:
it definitely didn't conflict or raise an issue at the time, i'll follow back on all of these import type things at the end since im iterating pretty quickly